group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/AssignmentTopicMetadata.java

```diff
@@ -25,12 +25,12 @@ public class AssignmentTopicMetadata {
     /**
      * The topic name.
      */
-    final String topicName;
+    private final String topicName;
```

Comment:
We should probably remove the `topicName` as we only use `topicId`.

Reply:
I think there's no harm in keeping it just in case we want to know what topic name is associated with what topicId wdyt?

Reply:
I would remove it because I don't really see the value of having it. All the interface works with topic ids now.